apis/Google.Cloud.Firestore/Google.Cloud.Firestore/ValueSerializer.cs

```diff
@@ -119,8 +119,10 @@ internal static Value Serialize(object value)
             GaxPreconditions.CheckNotNull(value, nameof(value));
             switch (value)
             {
+                case IDictionary<string, Value> map:
+                    return map.ToDictionary(pair => pair.Key, pair => pair.Value);
                 case IDictionary<string, object> map:
-                    return map.ToDictionary(pair => pair.Key, pair => Serialize(pair.Value));
+                    return map.ToDictionary(pair => pair.Key, pair => (pair.Value as Value) ?? Serialize(pair.Value));
```

Comment:
It might be better to instead update `Serialize` to have a case for `Value` that just returns the `value` parameter and then leave this line alone. That way any other type/collection that already is/contains `Value` instances will work similarly. @jskeet would know better if that's a good idea or not though.

Reply:
@evildour I am not sure about this too, so I would wait for @jskeet too